server/game/cards/15-DotE/BlueGraces.js

```diff
@@ -0,0 +1,26 @@
+const DrawCard = require('../../drawcard.js');
+
+class BlueGraces extends DrawCard {
+    setupCardAbilities(ability) {
+        this.action({
+            title: 'Shuffle card back into deck',
+            phase: 'challenge',
+            cost: ability.costs.removeSelfFromGame(),
+            target: {
+                activePromptTitle: 'Select a card',
+                cardCondition: card => (card.location === 'discard pile' || card.location === 'dead pile') &&

```

Comment:
The card text specifies your discard or dead pile, so controller needs to be checked.